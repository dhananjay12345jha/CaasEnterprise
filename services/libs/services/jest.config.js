module.exports = {
  displayName: "common-services",
  preset: "../../tools/jest.preset.js",
  globals: {
    "ts-jest": {
      tsconfig: "<rootDir>/tsconfig.spec.json",
    },
  },
  testEnvironment: "node",
  transform: {
    "^.+\\.[tj]sx?$": "ts-jest",
  },
  moduleFileExtensions: ["ts", "tsx", "js", "jsx"],
  coverageDirectory: "../../coverage/libs/services",
  coverageReporters: ["text", "cobertura", "json-summary", "html"],
  reporters: [
    "default",
    ["jest-junit", { outputDirectory: "./junit", outputName: "services.xml" }],
  ],
  coverageThreshold: {
    global: {
      branches: 85,
      functions: 85,
      lines: 85,
      statements: 85,
    },
  },
};
