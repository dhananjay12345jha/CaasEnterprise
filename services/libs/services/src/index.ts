export * from "./lib/errors/";
export * from "./lib/interfaces/";
export * from "./lib/psk/";
export * from "./lib/carts/";
export * from "./lib/orders/";
export * from "./lib/inventory/";
export * from "./lib/crm";
export * from "./lib/logger/";
export * from "./lib/payments/";
