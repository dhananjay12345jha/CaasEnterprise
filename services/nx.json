{
  "extends": "@nrwl/workspace/presets/core.json",
  "npmScope": "ei-services",
  "affected": {
    "defaultBase": "origin/develop"
  },
  "workspaceLayout": {
    "appsDir": "apps",
    "libsDir": "libs"
  },
  "cli": {
    "defaultCollection": "@nrwl/js",
    "packageManager": "yarn"
  },
  "tasksRunnerOptions": {
    "default": {
      "runner": "@nrwl/workspace/tasks-runners/default",
      "options": {
        "cacheableOperations": ["build", "lint", "test", "e2e"],
        "runtimeCacheInputs": [
          "echo BUILD_CONTEXT=${BUILD_CONTEXT:=local}",
          "echo DELIVERY_CONTEXT=${DELIVERY_CONTEXT:=sandbox}",
          "echo DELIVERY_STAGE=${DELIVERY_STAGE:=dev}"
        ]
      }
    }
  },
  "targetDefaults": {
    "build": {
      "dependsOn": ["^build"]
    }
  },
  "implicitDependencies": {
    "workspace.json": "*",
    "package.json": {
      "dependencies": "*",
      "devDependencies": "*"
    },
    "tsconfig.base.json": "*",
    ".eslintrc.json": "*",
    "nx.json": "*"
  }
}
