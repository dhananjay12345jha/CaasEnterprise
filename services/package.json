{
  "name": "@enterprise-integration/services",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "nx": "nx",
    "lint": "yarn nx workspace-lint && yarn nx lint",
    "build": "yarn nx build",
    "test": "yarn nx test",
    "test:unit": "yarn run test",
    "e2e": "yarn nx e2e",
    "format": "yarn nx format:write",
    "format:write": "yarn nx format:write",
    "format:check": "yarn nx format:check",
    "dep-graph": "yarn nx dep-graph",
    "serve": "yarn nx serve",
    "serve:dev": "yarn nx run-many --all --target=serve:local",
    "affected": "yarn nx affected",
    "affected:apps": "yarn nx affected:apps",
    "affected:libs": "yarn nx affected:libs",
    "affected:lint": "yarn nx affected:lint",
    "affected:info": "yarn nx affected --target=info",
    "affected:build": "yarn nx affected:build",
    "affected:test:feature": "yarn affected --target=e2e-feature",
    "affected:test:unit": "yarn nx affected:test",
    "affected:test:component": "yarn nx affected:e2e",
    "affected:test:performance": "yarn affected --target=insprint-performance",
    "affected:test:integration": "yarn affected --target=e2e-integration",
    "affected:version": "yarn version --patch --deferred",
    "affected:version:check": "yarn version check",
    "affected:version:apply": "yarn version apply",
    "affected:deploy": "yarn nx affected --target=deploy",
    "affected:destroy": "yarn nx affected --target=destroy",
    "workspaces:lint": "yarn nx workspace-lint"
  },
  "dependencies": {
    "@adyen/api-library": "^12.0.0",
    "@aws-sdk/client-dynamodb": "^3.105.0",
    "@aws-sdk/client-eventbridge": "^3.105.0",
    "@aws-sdk/client-sqs": "^3.112.0",
    "@aws-sdk/util-dynamodb": "^3.105.0",
    "@commercetools/platform-sdk": "^2.7.0",
    "@commercetools/sdk-client-v2": "^1.4.0",
    "@everymile-idp/brand-config-sdk": "file:../.idp-shim/dist/brand-config-sdk",
    "@everymile-schemas/order-created": "file:../contracts/private/jsonSchema/order/created",
    "@middy/core": "^2.5.3",
    "@middy/http-json-body-parser": "^2.5.3",
    "@middy/validator": "^3.1.0",
    "@types/fnv-plus": "^1.3.0",
    "ajv": "^8.11.0",
    "ajv-draft-04": "^1.0.0",
    "ajv-formats": "^2.1.1",
    "aws-lambda": "^1.0.7",
    "axios": "^0.27.2",
    "axios-retry": "^3.3.1",
    "fnv-plus": "^1.3.1",
    "iso-country-currency": "^0.6.0",
    "iso-datestring-validator": "^2.2.2",
    "newrelic": "^8.13.1",
    "node-fetch": "^2.6.1",
    "traverse": "^0.6.6",
    "tslib": "^2.4.0",
    "winston": "^3.8.2",
    "zod": "^3.17.3"
  },
  "devDependencies": {
    "@aws-sdk/client-eventbridge": "^3.110.0",
    "@enterprise-integration/mocks": "workspace:*",
    "@faker-js/faker": "^7.2.0",
    "@nrwl/cli": "13.10.1",
    "@nrwl/cypress": "13.10.1",
    "@nrwl/eslint-plugin-nx": "13.10.1",
    "@nrwl/jest": "13.10.1",
    "@nrwl/js": "13.10.1",
    "@nrwl/linter": "13.10.1",
    "@nrwl/node": "13.10.1",
    "@nrwl/tao": "13.10.1",
    "@nrwl/web": "13.10.1",
    "@nrwl/workspace": "13.10.1",
    "@serverless/typescript": "^3.21.0",
    "@types/aws-lambda": "^8.10.100",
    "@types/jest": "^27.5.1",
    "@types/lodash": "^4",
    "@types/newrelic": "^7.0.3",
    "@types/node": "^18.0.1",
    "@types/node-fetch": "^2.6.1",
    "@types/ramda": "^0.28.15",
    "@types/randomstring": "^1",
    "@types/serverless": "^3.12.7",
    "@types/traverse": "^0.6.32",
    "@types/uuid": "^8",
    "@typescript-eslint/eslint-plugin": "~5.20.0",
    "@typescript-eslint/parser": "~5.18.0",
    "@yarnpkg/esbuild-plugin-pnp": "^3.0.0-rc.2",
    "artillery": "^2.0.0-14",
    "axios": "^0.27.2",
    "axios-mock-adapter": "^1.21.2",
    "cypress": "^9.6.0",
    "dotenv": "^16.0.1",
    "esbuild": "^0.14.11",
    "eslint": "^8.0.0",
    "eslint-config-prettier": "^8.5.0",
    "jest": "27.2.3",
    "jest-fetch-mock": "^3.0.3",
    "jest-junit": "^13.2.0",
    "jest-util": "^28.1.3",
    "json-schema-to-ts": "^1.5.0",
    "kill-port": "^2.0.0",
    "mqtt": "^4.3.7",
    "newrelic": "^9.0.2",
    "nock": "^13.2.6",
    "nx": "13.10.1",
    "prettier": "^2.6.2",
    "ramda": "^0.28.0",
    "randomstring": "^1.2.2",
    "serverless": "^3.21.0",
    "serverless-esbuild": "^1.23.3",
    "serverless-iam-roles-per-function": "^3.2.0",
    "serverless-newrelic-lambda-layers": "^3.3.5",
    "serverless-offline": "^8.8.0",
    "serverless-offline-aws-eventbridge": "^1.6.6",
    "serverless-prune-plugin": "^2.0.1",
    "serverless-webpack": "^5.7.1",
    "start-server-and-test": "^1.14.0",
    "ts-jest": "27.0.5",
    "ts-loader": "^9.3.0",
    "ts-node": "^10.8.0",
    "tsconfig-paths": "^3.9.0",
    "tsconfig-paths-webpack-plugin": "^3.5.2",
    "typescript": "~4.5.5",
    "uuid": "^8.3.2",
    "webpack": "^5.72.0"
  }
}
