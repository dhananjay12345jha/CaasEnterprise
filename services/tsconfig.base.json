{
  "compileOnSave": false,
  "compilerOptions": {
    "sourceMap": true,
    "declaration": false,
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "importHelpers": true,
    "target": "ES2021",
    "lib": ["ES2021"],
    "module": "CommonJS",
    "skipLibCheck": true,
    "skipDefaultLibCheck": true,
    "baseUrl": ".",
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "paths": {
      "@ei-services/brand-config": ["libs/brand-config/src/index.ts"],
      "@ei-services/braze-service": ["libs/braze-service/src/index.ts"],
      "@ei-services/commerce-tools": ["libs/commerce-tools/src/index.ts"],
      "@ei-services/common/http": ["libs/common/http/src/index.ts"],
      "@ei-services/common/lambda": ["libs/common/lambda/src/index.ts"],
      "@ei-services/common/middleware": ["libs/common/middleware/src/index.ts"],
      "@ei-services/order-management-service": [
        "libs/order-management-service/src/index.ts"
      ],
      "@ei-services/services": ["libs/services/src/index.ts"],
      "@ei-services/shared/dlq/client": ["libs/shared/dlq/client/src/index.ts"],
      "@ei-services/shared/eventbridge/client": [
        "libs/shared/eventbridge/client/src/index.ts"
      ]
    }
  },
  "exclude": ["node_modules", "tmp"]
}
