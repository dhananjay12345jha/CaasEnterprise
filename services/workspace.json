{
  "version": 2,
  "projects": {
    "brand-config": "libs/brand-config",
    "braze-service": "libs/braze-service",
    "commerce-tools": "libs/commerce-tools",
    "common-http": "libs/common/http",
    "common-lambda": "libs/common/lambda",
    "common-middleware": "libs/common/middleware",
    "crm-newsletter-profile": "apps/crm/newsletter-profile",
    "crm-order-confirmation": "apps/crm/order-confirmation",
    "crm-order-dispatched": "apps/crm/order-dispatched",
    "crm-password-reset": "apps/crm/password-reset",
    "crm-user-registered": "apps/crm/user-registered",
    "order-management-service": "libs/order-management-service",
    "orders-async-events": "apps/orders/async-events",
    "orders-e2e": "apps/orders-e2e",
    "orders-status-changes": "apps/orders/status-changes",
    "orders-update-order-ct": "apps/orders/update-order-ct",
    "payments-e2e": "apps/payments-e2e",
    "payments-provider-callback": "apps/payments/provider-callback",
    "services": "libs/services",
    "shared-dlq-client": "libs/shared/dlq/client",
    "shared-eventbridge-client": "libs/shared/eventbridge/client",
    "stock-quantity-updates": "apps/stock/quantity-updates",
    "stock-update-quantity-ct": "apps/stock/update-quantity-ct"
  }
}
