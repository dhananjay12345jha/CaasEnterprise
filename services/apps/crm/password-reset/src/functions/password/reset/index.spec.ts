import { faker } from "@faker-js/faker";
import { default as resetPasswordHandler, HandlerEvent } from "./index";
import { Context as LambdaContext } from "aws-lambda/handler";
import { Logger } from "@ei-services/services";

const triggerMock = jest.fn();

jest.mock("./password-reset", () => {
  return {
    PasswordReset: jest.fn(() => {
      return {
        trigger: triggerMock,
      };
    }),
  };
});

describe("password reset handler", () => {
  beforeEach(() => jest.restoreAllMocks());

  it("should not proceed if brand id is missing", async () => {
    const event: HandlerEvent = {
      version: "1.0",
      id: faker.datatype.string(),
      "detail-type": "forgotpassword.reset.request",
      time: faker.datatype.string(),
      detail: {
        metadata: { "x-emc-ubid": null },
        payload: {
          email: faker.internet.email(),
          newCode: faker.datatype.uuid(),
          expiry: new Date().getTime(),
          customerId: faker.datatype.uuid(),
        },
      },
    };

    await expect(() =>
      resetPasswordHandler(event, {} as LambdaContext)
    ).rejects.toMatchSnapshot();
    expect(triggerMock).not.toHaveBeenCalled();
  });

  it("should call password reset trigger method with event payload", async () => {
    const dispatchId = faker.datatype.uuid();
    triggerMock.mockResolvedValue({ dispatch_id: dispatchId });

    const payload = {
      email: faker.internet.email(),
      newCode: faker.datatype.uuid(),
      expiry: new Date().getTime(),
      customerId: faker.datatype.uuid(),
    };

    const event: HandlerEvent = {
      version: "1.0",
      id: faker.datatype.string(),
      "detail-type": "forgotpassword.reset.request",
      time: faker.datatype.string(),
      detail: {
        metadata: { "x-emc-ubid": faker.datatype.uuid() },
        payload,
      },
    };

    await resetPasswordHandler(event, {} as LambdaContext);

    expect(triggerMock).toHaveBeenCalledWith(
      event.detail.metadata["x-emc-ubid"],
      payload
    );
  });

  it("should log an error when message trigger fails", async () => {
    triggerMock.mockRejectedValue(new Error("some-error"));

    const loggerSpy = jest.spyOn(Logger.prototype, "error");

    const event: HandlerEvent = {
      version: "1.0",
      id: faker.datatype.uuid(),
      "detail-type": "forgotpassword.reset.request",
      time: faker.datatype.string(),
      detail: {
        metadata: { "x-emc-ubid": faker.datatype.uuid() },
        payload: {
          email: faker.internet.email(),
          newCode: faker.datatype.uuid(),
          expiry: new Date().getTime(),
          customerId: faker.datatype.uuid(),
        },
      },
    };

    try {
      await resetPasswordHandler(event, {} as LambdaContext);
    } catch (err) {
      expect(loggerSpy).toHaveBeenCalledWith(
        "Error triggering password reset email: ",
        err
      );
    }
  });
});
