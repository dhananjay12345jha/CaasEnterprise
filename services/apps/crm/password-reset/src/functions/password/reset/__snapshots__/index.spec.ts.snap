// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`password reset handler should not proceed if brand id is missing 1`] = `
Array [
  Object {
    "code": "invalid_type",
    "expected": "string",
    "message": "Expected string, received null",
    "path": Array [
      "detail",
      "metadata",
      "x-emc-ubid",
    ],
    "received": "null",
  },
]
`;
